server/controllers/googleController.ts

```diff
@@ -30,13 +45,68 @@ const googleController: Record<string, RequestHandler> = {
    * Updates the global credentials for the googleapis
    * @param {string} req.query.code - The OAuth code sent to the redirect uri by google oauth api
    */
-  saveOAuthDetails: async (req, _res, next) => {
+  saveOAuthDetails: async (req: Request, res: Response, next: NextFunction) => {
     if (req.query.code !== undefined && typeof req.query.code === 'string') {
       const oAuthCode = req.query.code;
+      const sessionId = req.query.state as string;
+      const state = req.query.state as string;
+
+      console.log('Received OAuth code:', oAuthCode);
+      console.log('Received state from Google:', state);
+      console.log('Session ID from cookie:', sessionId);
+      console.log('Query Params:', req.query);
+
       try {
         const { tokens } = await oAuth2Client.getToken(oAuthCode);
-        oAuth2Client.setCredentials(tokens);
+        if (!tokens.access_token || !tokens.refresh_token) {
+          return next({
+            log: 'googleController.saveOAuthDetails: Missing access token or refresh token.',
+            status: 400,
+            message: {
+              err: 'OAuth token missing required fields.',
+            },
+          });
+        }
+
+        const accessToken: string = tokens.access_token as string;
+        const refreshToken: string | null = tokens.refresh_token
```

Comment:
You can use optional chaining syntax here:
`const refreshToken: string | undefined = tokens.refresh_token?`
or typescripty version of something similar:
`const refreshToken: string | null = tokens.refresh_token? ?? null`
(Its similar to ` y = x || 0`, where y gets assinged to x unless x is falsey, then it gets assigned to 0)